enhancing/enhancer.js

```diff
@@ -6,17 +7,77 @@ module.exports = {
 };
 
 function succeed(item) {
-  return { ...item };
+  if (typeof(item) !== 'object') {
+    return 'error: input must be an object';
+  }
+  let {name,durability,enhancement} = item;
+  if (!name || typeof(durability) !== 'number' || typeof(enhancement) !== 'number') {
+    return 'error: object must include name, durability, and enhancement properties';
+  }
+  
+  if (enhancement < 20) enhancement +=1;
+
+  const success = {
+    name:name,
+    durability:durability,
+    enhancement: enhancement
+  };
+  return { ...success };
 }
 
 function fail(item) {
-  return { ...item };
+  if (typeof(item) !== 'object') {
+    return 'error: input must be an object';
+  }
+  let {name,durability,enhancement} = item;
+  if (!name || typeof(durability) !== 'number' || typeof(enhancement) !== 'number') {
+    return 'error: object must include name, durability, and enhancement properties';
+  }
+
+  if (enhancement < 15) durability -=5;
+  if (enhancement >= 15) durability -=10;
+  if (enhancement > 16) enhancement -=1;
```

Comment:
extra edge case here: should you be accounting for the case that the durability might drop below zero due to this function?